internal/wrappers/cpp/core.cpp

```diff
@@ -0,0 +1,178 @@
+#include <exception>
+#include <stdexcept>
+#include <string>
+#include "kuzzle.hpp"
+#include <iostream>
+#include <vector>
+
+namespace kuzzleio {
+
+  KuzzleException::KuzzleException(int status, const std::string& error)
+    : std::runtime_error(error), status(status) {}
+
+  std::string KuzzleException::getMessage() const {
+    return what();
+  }
+
+  Kuzzle::Kuzzle(const std::string& host, options *opts) {
+    this->_kuzzle = (kuzzle*)malloc(sizeof(kuzzle));
+    kuzzle_new_kuzzle(this->_kuzzle, const_cast<char*>(host.c_str()), (char*)"websocket", opts);
+  }
+
+  Kuzzle::~Kuzzle() {
+    unregisterKuzzle(this->_kuzzle);
+    free(this->_kuzzle);
+  }
+
+  token_validity* Kuzzle::checkToken(const std::string& token) {
+    return kuzzle_check_token(_kuzzle, const_cast<char*>(token.c_str()));
+  }
+
+  char* Kuzzle::connect() {
+    return kuzzle_connect(_kuzzle);
+  }
+
+  bool Kuzzle::createIndex(const std::string& index, query_options* options) Kuz_Throw_KuzzleException {
+    bool_result *r = kuzzle_create_index(_kuzzle, const_cast<char*>(index.c_str()), options);
+    if (r->error != NULL)
+        throwExceptionFromStatus(*r);
+    return r->result;
+  }
+
+  json_object* Kuzzle::createMyCredentials(const std::string& strategy, json_object* credentials, query_options* options) Kuz_Throw_KuzzleException {
+    json_result* r = kuzzle_create_my_credentials(_kuzzle, const_cast<char*>(strategy.c_str()), credentials, options);
```

Comment:
potential memleak the memory allocated to this structure must be freed, either before returning the result, or after an exception is thrown